.github/workflows/actions.yml

```
@@ -0,0 +1,35 @@
+# Lint GitHub Actions for common security issues using zizmor.
+# Docs: https://woodruffw.github.io/zizmor
+
+name: lint-actions
+
+# Only run on PRs and the main branch.
+# Pushes to branches will only trigger a run when a PR is opened.
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install requirements
+        run: python -m pip install -r env/requirements-style.txt
+
+      - name: List installed packages
+        run: python -m pip freeze
+
+      - name: Lint GitHub Actions
+        run: make check-actions
```

@leouieda, I'm considering adding the GITHUB_TOKEN here to allow zizmor to check for vulnerabilities online (using the GitHub API). I think this is safe, but wanted to double check with you before applying this change.

```suggestion
      - name: Lint GitHub Actions
        run: make check-actions
        env:
          # Set GH_TOKEN to allow zizmor to check online vulnerabilities
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

My reasoning: the `GITHUB_TOKEN` [is passed to runners triggered from a forked repository](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow), but it has [read-only permissions](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) is such case. So no PR from external fork can gain writing permissions to our repo  by exploiting that token.

---

Sounds good. The matter is trusting zizmor to not steal tokens, right?

---

Yes. I think that's safe within our thread model.